grammars/c#.cson

```diff
@@ -84,12 +84,23 @@
   'builtinTypes':
     'patterns': [
       {
-        'match': '\\b(bool|byte|sbyte|char|decimal|double|float|int|uint|long|ulong|object|short|ushort|string|void|class|struct|enum|interface|dynamic)(\\?|\\*)?\\s'
-        'name': 'storage.type.cs'
+        'match': '\\b(bool|byte|sbyte|char|decimal|double|enum|float|int|uint|long|ulong|short|ushort|)\\s*(\\?|\\*|\\[\\s*\\])?\\s'
+        'name': 'storage.value.type.cs'
         'captures':
           '2':
             'name': 'punctuation.storage.type.modifier.cs'
       }
+      {
+        'match': '\\b(dynamic|object|string)\\s*(\\*|\\[\\s*\\])?\\s'
+        'name': 'storage.reference.type.cs'
+        'captures':
+          '2':
+            'name': 'punctuation.storage.type.modifier.cs'
+      }
+      {
+        'match': '\\b(void|class|struct|interface)\\s'
```

Comment:
Can this be made into `(?=\\s)` instead so that the ending space isn't tokenized?
Alternatively, reference by captures instead of the entire match.